### PR TITLE
make compile pass

### PR DESCRIPTION
**Pull request by megazone87**
1. add some missing functions
2. add cmake file
3. add header guards
4. fix typos
5. format code as google style

### CODE REVIEW

**Comment by megazone87** [comment]
The missing functions are declared only, not implemented.

File: lilcom/fixed_math.h [hunk 1]
```diff
@@ -283,7 +274,7 @@ void CopyFromScalar64(const Scalar64 *a, int i, Vector64 *y);
 void MulScalar64(const Scalar64 *a, const Scalar64 *b, Scalar64 *y);
 
 /* does: y := a.  Just copies all the elements of the struct. */
-extern inline void CopyScalar64(const Scalar64 *a, Scalar64 *y) {
+static inline void CopyScalar64(const Scalar64 *a, Scalar64 *y) {
```
Comment (danpovey) [inline lilcom/fixed_math.h:277]: You sure about 'static' here?  I thought when you declared these in a header it should be extern?
I am not so knowledgeable about this though.

**Comment by danpovey** [comment]
Great work!  Merging.

File: lilcom/fixed_math.h (hunk 1 above)
Comment (megazone87) [reply lilcom/fixed_math.h:277]: Hi dan,

About making "inline" static, i just find it work, so i used it. After you asked, I google it and find it's a tricky problem.
Ref: http://gudok.xyz/inline/

Reasons in brief:
1. Inline funtion is linked diffenently in C and C++, C: internal link, C++: external link. (In C, make definition of inline func be extern, will easily lead to the func duplicated defined)
2. In C, "-O0" don't inline the function actually, but "-O3" leads to real "inline" (except the func cannot be inlined). Thus, with "-O3", the inline func don't need to be "static".
3. But with fake inline("-O0"), the compiler ignores inline function body, thus that missing during linking.

Solutions for C:
1. add “static” to inline func
2. or, explicitly instantiate this function once, and make it extern. (It is a inferior solution.)

**Comment by danpovey** [comment]
Thanks a lot!

**Comment by danpovey** [comment]
After reading more about static/extern, it seems the only really
correct way would be to declare it as 'inline' in the header, say foo.h,
but use the preprocessor so that when included from exactly one translation
unit (say, foo.c), it shows up as "extern inline".

**Pull request merged**